### DISCUSSION TITLE
Is there an example of integrating a Neutralino frontend with a Java backend?

### DISCUSSION THREAD

**Discussion by ppazos** [neutralinojs/neutralinojs#1248] [General]
We have Java backend logic behind a Java CLI, so commands can be executed directly, and want to build a desktop app on top of that. We don't want to have a full REST API architecture for this communication between the front desktop app and the Java backend, which seems overkill.

Is there some tutorial or example for something like this?

Thanks!

**Comment by hschneider** [comment]
You can either call any binary file and read its output via shell execute:
https://neutralino.js.org/docs/api/os#osexeccommandcommand-options

Or you can setup a WebSocket client on Java's side and build a Neutralino Extension:
https://neutralino.js.org/docs/how-to/extensions-overview

Communicating thru WebSockets is closer to an API than the first solution.
